gapis/service/service.proto

```diff
@@ -671,6 +672,76 @@ message MemoryRange {
   uint64 size = 2;
 }
 
+message Metrics {
+  oneof metrics {
+    MemoryBreakdown memory_breakdown = 1;
+  }
+}
+
+// The description of the memory layout of the API state
+message MemoryBreakdown {
+  // The API ID used for this call.
+  path.API api = 1;
+  // The index to get the constants names for MemoryAllocation `flags`.
+  // A value of -1 indicates no flag names should be fetched.
+  int32 allocation_flags_index = 2;
+  // The memory allocations that are currently in use.
+  repeated MemoryAllocation allocations = 3;
+}
+
+// A single memory allocation
+message MemoryAllocation {
+  // The device this allocation was made on
+  uint64 device = 1;
+  // The memory type this allocation is from
+  uint32 memory_type = 2;
+  // Flags describing properties of the allocation, e.g. whether it's device
+  // local, whether it's host visible, etc.
+  uint32 flags = 3;
+  // Constant index for looking up the flag values with the path.ConstantSet endpoint
+  uint32 constant_set = 4;
+  // The API specific ID for this allocation
+  uint64 id = 5;
+  // The user-readable name for this allocation
+  string name = 6;
+  // This size of the memory allocated
+  uint64 size = 7;
+  // The regions of host memory this allocation has been mapped into
+  MemoryMapping mapping = 8;
+  // All buffers/images bound to this memory allocation
+  repeated MemoryBinding bindings = 9;
+}
+
+// A mapping from part of a memory allocation into host memory
+message MemoryMapping {
+  // The size in bytes of the mapping
+  uint64 size = 1;
+  // The offset of the mapping into the source allocation, in bytes
+  uint64 offset = 2;
+  // The address on the host that this memory is bound to
+  uint64 host_address = 3;
```

Comment:
I would prefer `mapped_address` rather than `host_address` since the host address doesn't have to necessarily be fixed for the entire applications lifetime (The driver is free to return different addresses for different mappings).